Tella/Scenes/Audio/AudioPlayer/Views/EditAudioView.swift

```diff
@@ -0,0 +1,246 @@
+        }
+        .frame(height: 30)
+        .padding(16)
+    }
+    
+    var timeLabelsView: some View {
+        HStack(spacing: 0) {
+            ForEach(editAudioViewModel.generateTimeLabels(), id: \.self) { time in
+                Text(time)
+                    .font(.system(size: 12, weight: .regular))
+                    .foregroundColor(.white.opacity(0.32))
+                if time != editAudioViewModel.generateTimeLabels().last {
+                    Spacer()
+                }
+            }
+        }.frame(width: kTrimViewWidth)
+            .padding([.top], 70)
+    }
+    
+    var displayTimeLabels: some View {
+        VStack {
+            Text(editAudioViewModel.currentTime)
+                .font(.custom(Styles.Fonts.regularFontName, size: 50))
+                .foregroundColor(.white)
+            Text(editAudioViewModel.audioPlayerViewModel.duration)
+                .font(.custom(Styles.Fonts.regularFontName, size: 14))
+                .foregroundColor(.gray)
+        }  .padding(.top, 70)
+    }
+    
+    
+    var controlButtonsView: some View {
+        HStack(spacing: 64) {
+            Button(action: { self.undo() }) {
+                Image("cancel.edit.file")
+                    .resizable()
+                    .frame(width: 50, height: 50)
+                    .foregroundColor(.gray)
+            }
+            
+            Button(action: { editAudioViewModel.handlePlayButton() }) {
+                Image(editAudioViewModel.playButtonImageName)
+                    .resizable()
+                    .frame(width: 50, height: 50)
+                    .foregroundColor(.gray)
+            }
+            
+        }
+        .padding(.top, 64)
+    }
+    
+    private func undo() {
+        editAudioViewModel.startTime = 0.0
+        editAudioViewModel.endTime = editAudioViewModel.timeDuration
+        leadingGestureValue = 0.0
+        trailingGestureValue = kTrimViewWidth
+    }
+    private func closeView() {
+        editAudioViewModel.isPlaying = false
+        if editAudioViewModel.isDurationHasChanged() {
+            isBottomSheetShown = true
+        }else  {
+            sheetManager.hide()
+            isPresented = false
+        }
+    }
+}
+
+
+
+private struct TrimAudioSliderView: View {
+    @Binding var value: Double
+    var range: ClosedRange<Double>
+    @Binding var gestureValue: Double
+    @Binding var shouldLimitScrolling: Bool
+    
+    @State private var isRightSlider = false //This variable to check if the slider starts from the left or the right side
+    private let kOffset = 3.0 //This is added to adjust the difference of the elipse in the trim line image
+    private let kLabelOffset = 15.0 //This constant is added to center the value label in the trim line
+    var body: some View {
+        GeometryReader { geometry in
+            ZStack(alignment: .leading) {
+                // Custom thumb with image
+                Image("edit.audio.trim.line").resizable()
```

Comment:
What do you think of splitting this view into small views and functions ?

    var body: some View {
        GeometryReader { geometry in
            ZStack(alignment: .leading) {
                
                sliderImage(in: geometry)
                    .gesture(dragGesture(in: geometry))
                
                valueLabel(in: geometry)
            }
        }
        .onAppear {
            isRightSlider = range.upperBound == value
        }
    }

    // Extract slider image view with offset calculation
    private func sliderImage(in geometry: GeometryProxy) -> some View {
        Image("edit.audio.trim.line")
            .resizable()
            .frame(width: 10, height: 200)
            .offset(x: calculateThumbOffset(in: geometry), y: -10)
    }
    
    // Extract value label with offset calculation
    private func valueLabel(in geometry: GeometryProxy) -> some View {
        Text("\(TimeInterval(value).toHHMMString())")
            .foregroundColor(Styles.Colors.yellow)
            .font(.custom(Styles.Fonts.regularFontName, size: 12))
            .offset(x: calculateLabelOffset(in: geometry), y: 105)
    }
    
    // Drag gesture with adjusted conditions for limit scrolling
    private func dragGesture(in geometry: GeometryProxy) -> some Gesture {
        DragGesture(minimumDistance: 0)
            .onChanged { dragValue in
                let newValue = Double(dragValue.location.x / geometry.size.width) * (range.upperBound - range.lowerBound) + range.lowerBound
                gestureValue = calculateThumbOffset(in: geometry)
                
                if shouldLimitScrolling {
                    if isRightSlider {
                        guard newValue >= value else { return }
                    } else {
                        guard newValue <= value else { return }
                    }
                }
                
                value = min(max(newValue, range.lowerBound), range.upperBound)
            }
            .onEnded { _ in
                gestureValue = calculateThumbOffset(in: geometry)
            }
    }
    
    // Calculate the offset for the thumb
    private func calculateThumbOffset(in geometry: GeometryProxy) -> CGFloat {
        CGFloat((value - range.lowerBound) / (range.upperBound - range.lowerBound)) * geometry.size.width - kOffset
    }
    
    // Calculate the offset for the value label
    private func calculateLabelOffset(in geometry: GeometryProxy) -> CGFloat {
        CGFloat((value - range.lowerBound) / (range.upperBound - range.lowerBound)) * geometry.size.width - kLabelOffset
    }